relayer/application_relayer.go

```diff
@@ -96,25 +95,29 @@ func newApplicationRelayer(
 		// Otherwise, the source subnet signs the message.
 		signingSubnet = sourceBlockchain.GetSubnetID()
 	}
+
+	sub := ticker.Subscribe()
+	checkpointManager := newCheckpointManager(logger, db, sub, relayerID)
+	checkpointManager.run()
```

Comment:
If `run()` should only be called once, maybe we should call it at the end of `newCheckpointManager`?

Reply:
I elected to keep them separate to give the caller greater control over the checkpoint manager processing loop. For instance, the tests in `checkpoint_test.go` only call `newCheckpointManager`. Requiring the unit tests to also run the processing loop seems like overkill, and potentially less stable.